compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -25,6 +26,8 @@ class CompilationUnit protected (val source: SourceFile) {
 
   def isJava: Boolean = source.file.name.endsWith(".java")
 
+  def isREPL: Boolean = source.file.name.startsWith(str.REPL_SESSION_LINE)
```

Comment:
Is it possible to generalize this a little, like `var isSuspendable = true`, and mutate the field for REPL compilation units?

I was thinking that maybe worksheets may have the same need.